pkg/agent/controller/service_export_client.go

```diff
@@ -57,13 +58,23 @@ func (c *ServiceExportClient) updateStatusConditions(name, namespace string, con
 func (c *ServiceExportClient) tryUpdateStatusConditions(name, namespace string, canReplace bool,
 	conditions ...mcsv1a1.ServiceExportCondition,
 ) {
+	findStatusCondition := func(conditions []mcsv1a1.ServiceExportCondition, condType mcsv1a1.ServiceExportConditionType,
```

Comment:
nit: Add a TODO here to eventually remove this code when we no longer support a version that still uses synced. Rest looks good.

Reply:
Done